src/stage1/analyze.cpp

```diff
@@ -4769,11 +4769,11 @@ Error type_is_nonnull_ptr2(CodeGen *g, ZigType *type, bool *result) {
     return ErrorNone;
 }
 
-static uint32_t get_async_frame_align_bytes(CodeGen *g) {
-    uint32_t a = g->pointer_size_bytes * 2;
-    // promises have at least alignment 8 so that we can have 3 extra bits when doing atomicrmw
-    if (a < 8) a = 8;
-    return a;
+uint32_t get_async_frame_align_bytes(CodeGen *g) {
+    // Due to how the frame structure is built the minimum alignment is the one
+    // of a usize (or pointer).
+    // label (grep this): [fn_frame_struct_layout]
+    return max(g->builtin_types.entry_usize->abi_align, target_fn_align(g->zig_target));
```

Comment:
Big question for @andrewrk, why is the compiler factoring in the target function alignment when calculating the minimum alignment for the frame type?

Reply:
Hmmm. I don't think it should be doing that. It should be determined by the largest aligned local that has to get spilled to the function frame, or usize (because of the fields the language needs to add to make async functions work).

Reply:
Indeed, the 2 * usize_align formula used previously left me quite confused at first.

> It should be determined by the largest aligned local that has to get spilled to the function frame

I'm definitely no expert in the coroutine implementation but ideally `anyframe`'s alignment is defined as the biggest among every frame in the compilation unit in order to be compatible with every single frame, no?